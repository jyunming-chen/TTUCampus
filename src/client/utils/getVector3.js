// @ts-check

import { Vector3 } from 'three';

/**
 * @param {number[]} xyOrXyzArray
 */
export default function getVector3(xyOrXyzArray) {
  if (xyOrXyzArray.length === 2) {
    const [x, y] = xyOrXyzArray;
    return new Vector3(x, 0, -y);
  }
  const [x, y, z] = xyOrXyzArray;
  return new Vector3(x, y, z);
}
