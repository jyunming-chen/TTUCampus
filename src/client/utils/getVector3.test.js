import getVector3 from './getVector3';

test('getVector3([1, 2])', () => {
  expect(getVector3([1, 2])).toMatchSnapshot();
});

test('getVector3([1, 2, 3])', () => {
  expect(getVector3([1, 2, 3])).toMatchSnapshot();
});
