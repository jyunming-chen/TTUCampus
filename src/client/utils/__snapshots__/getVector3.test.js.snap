// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getVector3([1, 2, 3]) 1`] = `
Vector3 {
  "x": 1,
  "y": 2,
  "z": 3,
}
`;

exports[`getVector3([1, 2]) 1`] = `
Vector3 {
  "x": 1,
  "y": 0,
  "z": -2,
}
`;
